src/uproot/_dask.py

```diff
@@ -168,6 +170,162 @@ def dask(
         raise NotImplementedError()
 
 
+class _PackedArgCallable:
```

Comment:
I'm wondering if uproot should use dask-awkward's implementation of `_PackedArgCallable` rather than rewriting it. What's the dependency situation right now? If a user wants to use `uproot.dask` do we already require them to have dask-awkward installed?

Reply:
> If a user wants to use `uproot.dask` do we already require them to have dask-awkward installed?

Currently no: if they call `uproot.dask` with `library="np"`, they don't need dask-awkward, but if `library="ak"`, then they do. I wasn't 100% sure of that decision.

You know, dask-awkward doesn't seem to be on conda-forge yet. That might be a blocker for some people. (Personally, I pip-install a minimal number of packages into a conda environment, if there's no conda equivalent.)

> I'm wondering if uproot should use dask-awkward's implementation of `_PackedArgCallable` rather than rewriting it.

I don't know; this is pretty small and straightforward:

```python
class _PackedArgCallable:
    """Wrap a callable such that packed arguments can be unrolled.
    Inspired by dask.dataframe.io.io._PackedArgCallable.
    """

    def __init__(self, func, args=None, kwargs=None, packed=False):
        self.func, self.args, self.kwargs, self.packed = func, args, kwargs, packed

    def __call__(self, packed_arg):
        if not self.packed:
            packed_arg = (packed_arg,)
        return self.func(*packed_arg, *(self.args or []), **(self.kwargs or {}))
```

But if we do want to pick it up from dask-awkward, it will have to have a non-underscored name.

Going in the other direction, if this came from Dask DataFrame and now both Dask-Awkward and Uproot need it, maybe it should be a public API function in Dask? (I know that that's not a quick change to make. But in principle?)

Reply:
The version in dask is inside of dask.dataframe (which if used will require `import pandas`); but now I'm thinking it should be moved out of `dask.dataframe` and into `dask`!